themes/default/content/docs/concepts/inputs-outputs/lifting.md

```diff
@@ -0,0 +1,316 @@
+
+{{% choosable language typescript %}}
+
+```typescript
+let certCertificate = new aws.acm.Certificate("cert", {
+    domainName: "example.com",
+    validationMethod: "DNS",
+});
+let certValidation = new aws.route53.Record("cert_validation", {
+    records: [
+        certCertificate.domainValidationOptions[0].resourceRecordValue
+    ],
+...
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+certificate = aws.acm.Certificate('cert',
+    domain_name='example.com',
+    validation_method='DNS'
+)
+
+record = aws.route53.Record('validation',
+    records=[
+        certificate.domain_validation_options[0].resource_record_value
+    ],
+...
+```
+
+{{% /choosable %}}
+
+{{% choosable language go %}}
+
+```go
+cert, err := acm.NewCertificate(ctx, "cert", &acm.CertificateArgs{
+    DomainName:       pulumi.String("example"),
+    ValidationMethod: pulumi.String("DNS"),
+})
+if err != nil {
+    return err
+}
+
+record, err := route53.NewRecord(ctx, "validation", &route53.RecordArgs{
+    Records: pulumi.StringArray{
+        // Notes:
+        // * `Index` looks up an index in an `ArrayOutput` and returns a new `Output`.
+        // * Accessor methods like `ResourceRecordValue` lookup properties of a custom struct `Output` and return a new `Output`.
+        // * `Elem` dereferences a `PtrOutput` to an `Output`, equivalent to `*`.
+        cert.DomainValidationOptions.Index(pulumi.Int(0)).ResourceRecordValue().Elem(),
+    },
+    ...
+})
+if err != nil {
+    return err
+}
+```
+
+{{% /choosable %}}
+
+{{% choosable language csharp %}}
+
+```csharp
+var cert = new Certificate("cert", new CertificateArgs
+{
+    DomainName = "example",
+    ValidationMethod = "DNS",
+});
+
+var record = new Record("validation", new RecordArgs
+{
+    // Notes:
+    // * `GetAt` looks up an index in an `Output<ImmutableArray<T>>` and returns a new `Output<T>`
+    // * There are not yet accessor methods for referencing properties like `ResourceRecordValue` on an `Output<T>` directly,
+    //   so the `Apply` is still needed for the property access.
+    Records = cert.DomainValidationOptions.GetAt(0).Apply(opt => opt.ResourceRecordValue!),
+});
+```
+
+{{% /choosable %}}
+
+{{% choosable language java %}}
+
+```java
+// Lifting is currently not supported in Java.
+```
+
+{{% /choosable %}}
+
+{{% choosable language yaml %}}
+
+```yaml
+resources:
+  cert:
+    type: aws:acm:Certificate
+    properties:
+      domainName: example
+      validationMethod: DNS
+  record:
+    type: aws:route53:Record
+    properties:
+      records:
+        # YAML handles inputs and outputs transparently.
+        - ${cert.domainValidationOptions[0].resourceRecordValue}
+```
+
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+This approach is easier to read and write and does not lose any important dependency information that is needed to properly create and maintain the stack. This approach doesn’t work in all cases, but when it does, it can be a great help.
```

Comment:
The updates of this page is not included in this round of edits, and I do plan to cover this because I actually know the answer now lol I've just been thinking about the best way of how to present that clearly.